server/repository/src/db_diesel/name_tag.rs

```diff
@@ -0,0 +1,75 @@
+use diesel::prelude::*;
+
+use crate::{
+    db_diesel::{
+        name_row::name::dsl as name_dsl,
+        name_tag_join::name_tag_join::dsl as name_tag_join_dsl,
+        name_tag_row::{name_tag, name_tag::dsl as name_tag_dsl},
+        store_row::store::dsl as store_dsl,
+    },
+    diesel_macros::apply_equal_filter,
+    repository_error::RepositoryError,
+    DBType, NameTagRow, StorageConnection,
+};
+
+use crate::EqualFilter;
+
+pub type NameTag = NameTagRow;
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct NameTagFilter {
+    pub store_id: Option<EqualFilter<String>>,
+}
+
+pub struct NameTagRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+pub type BoxedNameTagQuery = name_tag::BoxedQuery<'static, DBType>;
+
+impl<'a> NameTagRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        NameTagRepository { connection }
+    }
+
+    pub fn query(&self, filter: Option<NameTagFilter>) -> Result<Vec<NameTag>, RepositoryError> {
+        let query = Self::create_filtered_query(filter);
+
+        let result = query.load::<NameTag>(&self.connection.connection)?;
+
+        Ok(result)
+    }
+
+    pub fn create_filtered_query(filter: Option<NameTagFilter>) -> BoxedNameTagQuery {
+        let mut query = name_tag::table.into_boxed();
+
+        let Some(NameTagFilter { store_id }) = filter else {
+            return query;
+        };
+
+        if store_id.is_some() {
+            let mut name_tag_query = name_tag_join_dsl::name_tag_join
+                .left_join(name_dsl::name.left_join(store_dsl::store))
+                .into_boxed();
+
+            apply_equal_filter!(name_tag_query, store_id, store_dsl::id);
+
+            query = query.filter(
+                name_tag_dsl::id.eq_any(name_tag_query.select(name_tag_join_dsl::name_tag_id)),
+            );
```

Comment:
I don't think you need the join here do you ? Could just do this?
```suggestion
             let mut store_name_query = store_dsl::store.into_boxed();

            apply_equal_filter!(store_name_query, store_id, store_dsl::id);

            query = query
                .filter(name_tag_dsl::name.eq_any(store_name_query.select(store_dsl::name_id)));
```

Reply:
Actually I take this back, it breaks a test in 1508.5. `name` in `name_tag` is actually the `tag_name` I was thinking it was `name_id`. Based on this miss-understand I'm tempted to go back to calling it `tag_name` as that feels clearer to me...

Reply:
> Based on this miss-understand I'm tempted to go back to calling it tag_name as that feels clearer to me...

hmm, for some reason i like that name_tag, i think it's because if I think a tag for something else, like item, it sounds better as `item_tag`, vs `tag_item` and `tag_name`